### PR TITLE
[Python SDK] Add support for faucet auth token

### PR DESCRIPTION
**Pull request by banool**
### Description
This PR lets users set an auth token for bypassing faucet ratelimits.

### Test Plan
See that I'm ratelimited:
```
curl -X POST 'https://faucet.devnet.aptoslabs.com/mint?amount=10000000000&address=0xd0f523c9e73e6f3d68c16ae883a9febc616e484c4998a72d8899a1009e5a89d6';
```
```
Request rejected by 1 checkers
```

See that the bypassing works:
```
poetry run python -m examples.faucet-auth-token <auth_token>
```
```
=== Addresses ===
Alice: 0x0c0d9dfb9a94aa83535ce884e0f6a7c59a66b69d4841fa9770d6f6aefae8e8f6

=== Balance before ===
Alice: 0

=== Balance after ===
Alice: 20000000
```

### CODE REVIEW

File: ecosystem/python/sdk/aptos_sdk/async_client.py [hunk 1]
```diff
@@ -780,19 +780,22 @@ class FaucetClient:
     base_url: str
     rest_client: RestClient
 
-    def __init__(self, base_url: str, rest_client: RestClient):
+    def __init__(self, base_url: str, rest_client: RestClient, auth_token: str = None):
```
Comment (davidiw) [inline ecosystem/python/sdk/aptos_sdk/async_client.py:783]: hopefully this actually fails the python liniting, but this should be `Optional[str]`

File: ecosystem/python/sdk/aptos_sdk/async_client.py [hunk 2]
```diff
@@ -780,19 +780,22 @@ class FaucetClient:
     base_url: str
     rest_client: RestClient
```
Comment (davidiw) [inline ecosystem/python/sdk/aptos_sdk/async_client.py:782]: need a `auth_token: Optional[str]` here

File: ecosystem/python/sdk/aptos_sdk/async_client.py [hunk 3]
```diff
@@ -780,19 +780,22 @@ class FaucetClient:
...
     async def fund_account(self, address: AccountAddress, amount: int):
         """This creates an account if it does not exist and mints the specified amount of
         coins into that account."""
-        response = await self.rest_client.client.post(
-            f"{self.base_url}/mint?amount={amount}&address={address}"
-        )
+        request = f"{self.base_url}/mint?amount={amount}&address={address}"
+        headers = {}
+        if self.auth_token:
+            headers["Authorization"] = f"Bearer {self.auth_token}"
```
Comment (davidiw) [inline ecosystem/python/sdk/aptos_sdk/async_client.py:797]: a more suitable approach might be to place the headers into the object during construction (in `__init__`) so we don't reconstruct them here each time.

File: ecosystem/python/sdk/examples/faucet-auth-token.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Copyright © Aptos Foundation
```
Comment (davidiw) [inline ecosystem/python/sdk/examples/faucet-auth-token.py:1]: I don't think we should commit this

**Review by davidiw — COMMENTED** [review]
thanks for doing this!

**Review by gregnazario — APPROVED** [review]

**Review by 0xmaayan — APPROVED** [review]
LGTM

**Pull request merged**